qiskit_braket_provider/providers/adapter.py

```diff
@@ -350,6 +356,36 @@ def aws_device_to_target(device: AwsDevice) -> Target:
     return target
 
 
+def decompose_fully(circuit: QuantumCircuit) -> QuantumCircuit:
+    """Returns a qiskit circuit which is decomposed as much as possible.
+    Args:
+            circuit (QuantumCircuit): Qiskit Quantum Cricuit
+
+    Returns:
+        QuantumCircuit: decomposed Qiskit circuit
+    """
+    old_circuit, decomposed_circuit = None, circuit
+    translatable_gates = set(qiskit_gate_names_to_braket_gates.keys())
+    translatable_gates = translatable_gates.union({"measure", "barrier"})
+    # decompose quantum circuit
+    while not (
+        {gate.name for gate, _, _ in decomposed_circuit.data}.issubset(
+            translatable_gates
+        )
+    ):
+        old_circuit = decomposed_circuit
+        decomposed_circuit = old_circuit.decompose()
```

Comment:
Is this always guaranteed to terminate?

Reply:
Reasonably this should terminate for circuits that are composed of standard gates, and measure/barrier instructions since the adapter passes the standard gates and paramterized circuit test. 

This may pose a potential bug though if my assumption is not correct and I can mitigate it by adding a condition where if decomposing the gate does nothing then the loop breaks.